src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs

```diff
@@ -49,7 +49,9 @@ internal static string GenerateKey(ViewLocationExpanderContext context)
         {
             var keyBuilder = new StringBuilder();
             var routeValues = context.ActionContext.RouteData.Values;
-            var controller = routeValues.GetValueOrDefault<string>(RazorViewEngine.ControllerKey);
+            var controller = RouteValueUtility.GetNormalizedRouteValue(
```

Comment:
Why not just have this be a static on RazorViewEngine - you already have the keys as statics on that class.